overlord/snapstate/snapstate.go

```diff
@@ -853,7 +853,10 @@ func (bk byKind) Less(i, j int) bool {
 
 	iRevOrd := kindRevOrder[bk[i].Type]
 	jRevOrd := kindRevOrder[bk[j].Type]
-	return iRevOrd >= jRevOrd
+	if iRevOrd != jRevOrd {
+		return iRevOrd >= jRevOrd
```

Comment:
Why >= and not < (sorry if this question is dumb)

Reply:
Because of a nifty trick with `kindRevOrder` priority map. The higher priority items eg. `snap.TypeOS` has priority 2, while `snap.TypeApp` which is not listed at all and thus `kindRevOrder[snap.TypeApp] => 0`. Meaning, `i` orders before `j` if `prio_i > prio_j`.

Now that I look at it, I should replace `>=` with `>`.